src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedCmsTests.cs

```diff
@@ -500,6 +500,87 @@ public static void AddFirstSigner_RSA(SubjectIdentifierType identifierType, bool
             cms.CheckSignature(true);
         }
 
+        [Fact]
+        public static void AddSigner_RSA_EphemeralKey()
+        {
+            using (RSA rsa = RSA.Create())
+            using (X509Certificate2 publicCertificate = Certificates.RSA2048SignatureOnly.GetCertificate())
+            using (X509Certificate2 certificateWithKey = Certificates.RSA2048SignatureOnly.TryGetCertificateWithPrivateKey(exportable: true))
+            {
+                if (certificateWithKey == null)
+                {
+                    return;
+                }
+                RSAParameters rsaParameters;
+                using (RSA privateKey = certificateWithKey.GetRSAPrivateKey(), exportableKey = privateKey.MakeExportable())
+                {
+                    rsaParameters = exportableKey.ExportParameters(true);
+                }
+                rsa.ImportParameters(rsaParameters);
+                using (X509Certificate2 certWithEphemeralKey = publicCertificate.CopyWithPrivateKey(rsa))
+                {
+                    ContentInfo content = new ContentInfo(new byte[] { 1, 2, 3 });
+                    SignedCms cms = new SignedCms(content, false);
+                    CmsSigner signer = new CmsSigner(certWithEphemeralKey);
+                    cms.ComputeSignature(signer);
+                }
+            }
+        }
+
+        [Fact]
+        public static void AddSigner_DSA_EphemeralKey()
+        {
+            using (DSA dsa = DSA.Create())
+            using (X509Certificate2 publicCertificate = Certificates.Dsa1024.GetCertificate())
+            using (X509Certificate2 certificateWithKey = Certificates.Dsa1024.TryGetCertificateWithPrivateKey(exportable: true))
+            {
+                if (certificateWithKey == null)
+                {
+                    return;
+                }
+                DSAParameters dsaParameters;
+                using (DSA privateKey = certificateWithKey.GetDSAPrivateKey(), exportableKey = privateKey.MakeExportable())
+                {
+                    dsaParameters = exportableKey.ExportParameters(true);
```

Comment:
Why not just go straight for `dsa.ImportParameters(exportableKey.ExportParameters(true));` and save on the local?